sdk/python/tests/test_BaseValue.py

```diff
@@ -38,53 +39,79 @@ class BaseValueTest(ComparisonTestUtils, unittest.TestCase):
-
-	def test_signed_base_value_caps_values(self):
-		# Act + Assert: positive wrap around to negative
-		self.assertEqual(126, BaseValue(1, 126, None, True).value)
-		self.assertEqual(127, BaseValue(1, 127, None, True).value)
-		self.assertEqual(-128, BaseValue(1, 128, None, True).value)
-		self.assertEqual(-127, BaseValue(1, 129, None, True).value)
-
-		self.assertEqual(-127, BaseValue(1, 129 + 1024, None, True).value)
-
-		# - negative wrap around to positive
-		self.assertEqual(-127, BaseValue(1, -127, None, True).value)
-		self.assertEqual(-128, BaseValue(1, -128, None, True).value)
-		self.assertEqual(127, BaseValue(1, -129, None, True).value)
-		self.assertEqual(126, BaseValue(1, -130, None, True).value)
-
-		self.assertEqual(126, BaseValue(1, -130 - 1024, None, True).value)
-
-	def _equality_and_inequality_are_supported(self, descriptor):
+	def test_cannot_create_unsigned_with_values_outside_range(self):
+		# Arrange:
+		test_cases = [
+			(1, -1), (1, 0x100),
+			(2, -1), (2, 0x1_0000),
+			(4, -1), (4, 0x1_0000_0000),
+			(8, -1), (8, 0x1_0000_0000_0000_0000)
+		]
+
+		for (size, raw_value) in test_cases:
+			# Act + Assert:
+			with self.assertRaises(ValueError):
+				BaseValue(size, raw_value, None, False)
+
+	def test_can_create_unsigned_with_values_inside_range(self):
+		# Arrange:
+		test_cases = [
+			(1, 0), (1, 0x24), (1, 0xFF),
+			(2, 0), (2, 0x243F), (2, 0xFFFF),
+			(4, 0), (4, 0x243F_6A88), (4, 0xFFFF_FFFF),
+			(8, 0), (8, 0x243F_6A88_85A3_08D3), (8, 0xFFFF_FFFF_FFFF_FFFF)
+		]
+
+		for (size, raw_value) in test_cases:
+			# Act:
+			value = BaseValue(size, raw_value, None, False)
+
+			# Assert:
+			self.assertEqual(size, value.size)
+			self.assertEqual(raw_value, value.value)
+
+	def test_cannot_create_signed_with_values_outside_range(self):
+		# Arrange:
+		test_cases = [
+			(1, -0x81), (1, 0x80),
+			(2, -0x8001), (2, 0x8000),
+			(4, -0x8000_0001), (4, 0x8000_0000),
+			(8, -0x8000_0000_0000_0001), (8, 0x8000_0000_0000_0000)
+		]
+
+		for (size, raw_value) in test_cases:
+			# Act + Assert:
+			with self.assertRaises(ValueError):
+				BaseValue(size, raw_value, None, True)
+
+	def test_can_create_signed_with_values_inside_range(self):
+		# Arrange:
+		test_cases = [
+			(1, -0x80), (1, 0x24), (1, 0x7F),
+			(2, -0x8000), (2, 0x243F), (2, 0x7FFF),
+			(4, -0x8000_0000), (4, 0x243F_6A88), (4, 0x7FFF_FFFF),
+			(8, -0x8000_0000_0000_0000), (8, 0x243F_6A88_85A3_08D3), (8, 0x7FFF_FFFF_FFFF_FFFF)
+		]
+
+		for (size, raw_value) in test_cases:
+			# Act:
+			value = BaseValue(size, raw_value, None, True)
+
+			# Assert:
+			self.assertEqual(size, value.size)
+			self.assertEqual(raw_value, value.value)
+
+	def _equality_and_inequality_are_supported(self, descriptor, is_signed):
 		descriptor = EqualityTestDescriptor(
 			descriptor.untagged,
 			descriptor.tagged,
-			lambda: random.randint(0, 2**64 - 1),
+			lambda: random.randint(0, 2 ** (32 if is_signed else 64) - 1),
```

Comment:
why 32 for signed?

Reply:
64 is too big (greater than signed max), i could change range